README.md

```diff
@@ -1,58 +1,148 @@
+```ruby
+def edit
+end
 
-What does update need to do in the action?
+def update
+end
+```
 
-similar to edit, the first step is to simply find the post we're talking about.
+And then create the edit view template `touch app/views/posts/edit.html.erb`
 
-once we found it, how can we update that instance?
+In the future we will call the form partial to integrate the form, but for this lesson let's just copy and paste the form code from the `new` form:
 
-manually, or let's use .update(:title => params[:post_title]). hm, that mass assignment seems like it could be easier...why can't we just pass params in entirely? what would we need to format params as in order to simply pass it or one of it's keys entirely to update?
+```ERB
+<h3>Post Form</h3>
 
-so we manually assign and save, or use update, and then we can redirect to post show and that's the edit / update flow.
+<%= form_tag posts_path do %>
+  <label>Post title:</label><br>
+  <%= text_field_tag :title %><br>
 
-another thing we don't like is that form is really dirty. lots of repitition. Is there a better way to build forms and controller actions that process them then doing all this reptitive work when there is clearly a pattern here?
+  <label>Post Description</label><br>
+  <%= text_area_tag :description %><br>
 
-what's the pattern - the post is about fields for the @post object and the udate logic corresponds to the fields in the form.
+  <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
+  
+  <%= submit_tag "Submit Post" %>
+<% end %>
+```
 
-Bonus: Are we updating the entire post or only the fields the change?
+If you open the browser and go to the `edit` page it will now display the form, but you may have noticed a pretty big flaw, it doesn't load the record's data into the form, there are a few tasks that we'll need to do in order to implement this behavior, first let's have our `edit` action store the `post` record in an instance variable:
 
-in the event you only edit the title, does the post content get updated if we are always assigning all fields? dirty_attributes like content_changed? internalized in the sql for save/update (is that true?)
+```ruby
+def edit
+  @post = Post.find(params[:id])
+end
+```
+
+Now that the `edit` view template will have access to the `@post` object, we need to refactor the form so that it auto fills the form fields with the data from the `post` record, this is done below:
+
+```ERB
+<% # app/views/posts/edit.html.erb %>
+
+...
+<%= text_field_tag :title, @post.title %><br>
+...
+<%= text_area_tag :description, @post.description %><br>
+...
+```
+
+This will now populate the form, but if you tried to submit the form you may notice that it's redirecting to the show page and not changing the values. We need to first make a change to the form so that it knows what route the data should be passed through as well as what HTTP verb needs to be called, make the following change to the `form_tag` line:
```

Comment:
Point out that the path for update is the same post_path, just the method is different. Which makes sense because we are working with the same resource, just doing a different action.

Ugh I love the HTTP protocol. Tim Berners-Lee is such a genius